src/components/AdditionalInfo/AdditionalInfo.unit.spec.jsx

```diff
@@ -7,38 +7,145 @@ import AdditionalInfo from './AdditionalInfo.jsx';
 import ExpandingGroup from '../ExpandingGroup/ExpandingGroup';
 
 describe('<AdditionalInfo/>', () => {
-  let wrapper;
-  // for the sake of more tests could be written
-  // declared beforeEach and assigned to wrapper
-  beforeEach(() => {
-    wrapper = mount(<AdditionalInfo triggerText="test" />).setState({
-      open: true,
+  describe('component', () => {
+    let wrapper;
+    // for the sake of more tests could be written
+    // declared beforeEach and assigned to wrapper
+    beforeEach(() => {
+      wrapper = mount(<AdditionalInfo triggerText="test" />).setState({
+        open: true,
+      });
     });
-  });
 
-  it('should render', () => {
-    expect(wrapper.text()).to.contain('test');
-    expect(wrapper.find('h4').length).to.equal(0);
-  });
-  it('should pass aXe check', () =>
-    axeCheck(<AdditionalInfo triggerText="test" />));
-  it('should render title container as heading', () => {
-    wrapper = mount(
-      <AdditionalInfo tagName={'h4'} triggerText="test" />,
-    ).setState({
-      open: true,
+    it('should render', () => {
+      expect(wrapper.text()).to.contain('test');
+      expect(wrapper.find('h4').length).to.equal(0);
+    });
+    it('should pass aXe check', () =>
+      axeCheck(<AdditionalInfo triggerText="test" />));
+    it('should render title container as heading', () => {
+      wrapper = mount(
+        <AdditionalInfo tagName={'h4'} triggerText="test" />,
+      ).setState({
+        open: true,
+      });
+      expect(wrapper.find('h4').length).to.equal(1);
+      wrapper.unmount();
+    });
+    it('renders both children when open is true', () => {
+      const first = wrapper.find('ExpandingGroup').props();
+      expect(first.open).to.be.true;
+      return axeCheck(
+        <ExpandingGroup open={first.open}>
+          <div className="child1" />
+          <div className="child2" />
+        </ExpandingGroup>,
+      );
     });
-    expect(wrapper.find('h4').length).to.equal(1);
-    wrapper.unmount();
   });
-  it('renders both children when open is true', () => {
-    const first = wrapper.find('ExpandingGroup').props();
-    expect(first.open).to.be.true;
-    return axeCheck(
-      <ExpandingGroup open={first.open}>
-        <div className="child1" />
-        <div className="child2" />
-      </ExpandingGroup>,
-    );
+
+  describe('analytics event', () => {
+    it('should include the correct expand action', () => {
+      const handleExpandAnalyticsEvent = e => {
+        expect(e.detail).to.deep.eql({
+          componentName: 'AdditionalInfo',
+          action: 'expand',
+          details: {
+            triggerText: 'This is test triggerText.',
+          },
+        });
+      };
+
+      global.document.body.addEventListener(
+        'component-library-analytics',
+        handleExpandAnalyticsEvent,
+      );
```

Comment:
I'm back to being paranoid about what happens if, for some reason, the component fails to trigger the custom event. This `handleExpandAnalyticsEvent` wouldn't be called, and the `expect` in it would never be run. The test would then succeed when it should fail.

What do you think of something like this?
```js
const handleExpandAnalyticsEvent = sinon.spy(e => { expect(...); });
// Add the listener
// Do the thing
// Remove the listener
expect(handleExpandAnalyticsEvent.called).to.be.true;
```

This way, we not only make sure the event is what we expect it to be, but also that the event was fired at all.

Potentially, this could be its own separate test, but I like the idea of adding it into every test just in case. Since it ensures the validity of the test itself.

Reply:
Ha! Just saw @bkjohnson had the same idea in [your other PR](https://github.com/department-of-veterans-affairs/component-library/pull/41#discussion_r581464447). :joy: